### PR TITLE
PR for Reviewing the current progress of the App

### PR DESCRIPTION
**Pull request by emugaya**
This PR is to allow the TTL and Mentor to review the current on the Ridesharing APP.

### CODE REVIEW

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def google_oauth2
+    @user = User.from_omniauth(request.env['omniauth.auth'])
+    if @user.persisted?
+    flash[:notice] = I18n.t 'devise.omniauth_callbacks.success', kind: 'Google'
```
Comment (morriskimani) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Please indent this appropriately

File: spec/controllers/ride_matches_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe RideMatchesController, type: :controller do
+  let!(:user) { create :user }
+  let!(:ride) { create :ride }
+  let!(:second_user) {create :second_user}
+
+  describe 'MATCH / UNMATCH Ride' do
+    before :each { @ride = Ride.first }
```
Comment (morriskimani) [inline spec/controllers/ride_matches_controller_spec.rb]: Is this really necessary? I believe we can access `ride` in places where you are using `@ride`

File: spec/controllers/ride_matches_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  
+    it 'matches a user to a ride offer or request' do
+      sign_in second_user
+      expect{
+        post :create, :params => { ride_id: @ride.id }
```
Comment (morriskimani) [inline spec/controllers/ride_matches_controller_spec.rb]: Prefer newer style for Hashes where keys are symbols 

`post :create, params: { ride_id: @ride.id }`

File: spec/support/controller_shared_examples.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# controller_shared_examples.rb
+
+RSpec.shared_examples 'GET #index' do |controller, ride_type_param|
+  describe controller, type: :controller do
```
Comment (morriskimani) [inline spec/support/controller_shared_examples.rb]: Consider testing that the correct values are set for `@page_header` using assigns...

File: spec/controllers/rides_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+require_relative '../support/controller_shared_examples.rb'
+
+RSpec.describe RidesController, type: :controller do
+  let!(:user) { create :user }
+  let!(:second_user) { create :second_user }
+
+  context 'when viewing different ride categories' do
+    it_behaves_like 'GET #index', RidesController, 'Offers', 'Current Ride Offers'
+    it_behaves_like 'GET #index', RidesController, 'Requests', 'Current Ride Offers'
```
Comment (morriskimani) [inline spec/controllers/rides_controller_spec.rb]: I think you meant `'Current Ride Requests'`

File: spec/controllers/rides_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        sign_in user
+        expect {
+          post :create,  :params => { ride:  attributes_for(:invalid_ride)}
+        }.to_not change(Ride, :count)
+      end
+    end
+  end
+
+  describe 'PATCH/PUT #update' do
+    let!(:ride) { create :ride }
```
Comment (morriskimani) [inline spec/controllers/rides_controller_spec.rb]: Do we need the ride to be created immediately? Why not just `let` without the `!`

File: spec/support/controller_shared_examples.rb (hunk 4 above)
Comment (emugaya) [reply spec/support/controller_shared_examples.rb]: @morriskimani I have added this test for correct values of `@page_header`

File: spec/controllers/rides_controller_spec.rb (hunk 5 above)
Comment (emugaya) [reply spec/controllers/rides_controller_spec.rb]: @morriskimani This has been changed.

File: spec/factories/ride_factories.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
+FactoryBot.define do
```
Comment (morriskimani) [inline spec/factories/ride_factories.rb]: Suggestion about file naming. Consider having this file named `ride.rb`. Just like you name your models.

File: spec/models/ride_model_spec.rb [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  context 'when retrieving Rides' do
+    it_behaves_like 'retrieving all model items', Ride, 1
+  end
+
+  context 'when retrieving Ride related objects' do
+    it_behaves_like 'items belonging to object', Ride, 'users'
+    it_behaves_like 'items belonging to object', Ride, 'ride_matches'
+
+    it 'returns a ride match for a particular user' do
+      ride = Ride.first
```
Comment (morriskimani) [inline spec/models/ride_model_spec.rb]: Why are you reassigning `ride`. We already defined that using `let` at the very top. I believe that's what you want to use

File: spec/models/ride_model_spec.rb [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+      user_id = User.second.id
```
Comment (morriskimani) [inline spec/models/ride_model_spec.rb]: same here. I believe what you are looking for is `second_user.id`

**Pull request closed without merging**